### DISCUSSION TITLE
Elasticsearch host error

### DISCUSSION THREAD

**Discussion by success4us** [jertel/elastalert2#155] [Q&A] [answered]
Hi, I have an elk setup on GKE and I'm trying to use elastalert for sending notifications on error logs. I'm new to elastalert, I just followed your README doc on this URL https://github.com/jertel/elastalert2/tree/master/chart/elastalert2. after deploying the elastalert2 helm chart, elastalert pod is failing to connect with the existing elasticsearch deployment.

I did not change any of the values from values.yaml except for rules, because the existing elasticsearch deployment's service name is elasticsearch itself and the port is 9200 which are same as values.yaml .

the error log from elastalert pod is:

Traceback (most recent call last):
  File "/usr/local/bin/elastalert-create-index", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/elastalert/create_index.py", line 218, in main
    host = args.host if args.host else input('Enter Elasticsearch host: ')
EOFError: EOF when reading a line
Enter Elasticsearch host:

Could you tell me what I'm doing wrong here?

thanks in advance.

**Comment by jertel** [comment, marked as answer]
It appears that the pod didn't get the config file mapped correctly. Use kubectl to exec into the elastalert2 pod and inspect the /opt structure. There should be a file located at `/opt/elastalert/config.yaml` with the Elastic host/port etc. 

PS, I moved this to a discussion since it doesn't follow the definition of an Elastalert2 issue.

**Reply by success4us** [reply]
Hi, I am not able to exec into the pod because the pod is crashing before it's getting started. the error from the pods is 
CrashLoopBackOff: back-off 1m20s restarting failed container=elastalert pod=elastalert-elastalert2-58c69c5895-hmk74_kube-logging(1488e973-42d8-43cd-ae22-02b3f74291a6)

and the logs on the container is 

Traceback (most recent call last):
  File "/usr/local/bin/elastalert-create-index", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/elastalert/create_index.py", line 218, in main
    host = args.host if args.host else input('Enter Elasticsearch host: ')
EOFError: EOF when reading a line
Enter Elasticsearch host:

**Reply by ferozsalam** [reply]
Try changing the `command` for the elastalert container to `["sleep", "10000"]` and redeploying. This should allow you to exec into the container and examine its state without the container trying to start Elastalert.

**Reply by success4us** [reply]
The config.yaml is there at /opt/elastalert

the content of the file is

---
rules_folder: /opt/elastalert/rules
scan_subdirectories: false
run_every:
  minutes: 1
buffer_time:
  minutes: 15
es_host: elasticsearch
es_port: 9200
writeback_index: elastalert
use_ssl: False
verify_certs: True
alert_time_limit:
  minutes: 2880

I even tried "ping elasticsearch:9200"

I got the output 

PING elasticsearch:9200 (10.2.1.35): 56 data bytes
64 bytes from 10.2.1.35: seq=0 ttl=62 time=2.037 ms
64 bytes from 10.2.1.35: seq=1 ttl=62 time=0.549 ms
64 bytes from 10.2.1.35: seq=2 ttl=62 time=0.518 ms
64 bytes from 10.2.1.35: seq=3 ttl=62 time=0.435 ms
^C
--- elasticsearch:9200 ping statistics ---
4 packets transmitted, 4 packets received, 0% packet loss
round-trip min/avg/max = 0.435/0.884/2.037 ms

**Reply by ferozsalam** [reply]
Interesting, the error message suggests that this is falling into the conditional [here](https://github.com/jertel/elastalert2/blob/master/elastalert/create_index.py#L213), which in turn suggests that `os.path` is unable to open the config file.

**Reply by ferozsalam** [reply]
I have discovered the source of your problem. Are you using the latest Helm chart version? In a [recent change](https://github.com/jertel/elastalert2/commit/06818877e2e25144f3c3bfa35362081f9d9c4130), the default location of the configuration file has been moved.

However [the Helm chart](https://github.com/jertel/elastalert2/blob/master/chart/elastalert2/values.yaml#L28) pins an earlier Docker image, which would be looking for the configuration file elsewhere.

Could you try changing the `tag:` on your chart to `latest` and seeing if that resolves the issue?

**Reply by success4us** [reply]
Yeah, It is working now with the latest tag. Thanks

**Reply by Raks2110** [reply]
Hi all i am facing the same error , i dint use helm it was manual deployment and there is no scope of using helm in our organisation ...how to rectify it any help

**Reply by ferozsalam** [reply]
Hey @Raksniru if you aren't using Helm then your issue is unrelated, and probably means you're not setting the `es_host` config value or the `ES_HOST` environment variable correctly. Check the documentation [here](https://elastalert2.readthedocs.io/en/latest/elastalert.html#configuration) and [here](https://elastalert2.readthedocs.io/en/latest/elastalert.html#running-elastalert) to confirm you're running ElastAlert as expected.

If you're still having issues this is potentially something for a new discussion topic.

**Reply by Raks2110** [reply]
I have set the Host variable now , the issue is when i exec into the container there i could not find "config.yaml" this file thats why my pod is crashing

**Reply by ferozsalam** [reply]
Have you mounted a configuration file into your container? See the README instructions [here](https://github.com/jertel/elastalert2#docker). In any case, could you open a new discussion topic for this? This thread shouldn't become the go-to point for all issues with setting the host variable.
